### PR TITLE
Conditional Density Estimation notebook

### PR DESCRIPTION
**Pull request by vdutor**
Notebook building and fitting a deep (two layer) latent variable model using VI. No changes to the core of GPflux are required but careful setting of fitting options is necessary. For example, it is important to set `shuffle` to `False` and `batch_size` to the number of datapoints to have correct optimisation of the latent variables.

### CODE REVIEW

File: docs/notebooks/intro.py [hunk 1]
```diff
@@ -43,6 +43,7 @@ def motorcycle_data():
     df = pd.read_csv("./data/motor.csv", index_col=0)
     X, Y = df["times"].values.reshape(-1, 1), df["accel"].values.reshape(-1, 1)
     Y = (Y - Y.mean()) / Y.std()
+    X /= X.max()
```
Comment (awav) [inline docs/notebooks/intro.py:46]: How is this relevant to the problem?

**Comment by awav** [comment]
Can you summarise what did you add to fix the issue?

**Comment by st--** [comment]
@vdutor was it intentional that you did not save it as jupytext .py instead of .ipynb format?

File: docs/notebooks/intro.py (hunk 1 above)
Comment (vdutor) [reply docs/notebooks/intro.py:46]: Rescales inputs between [0,1] instead of the [0,60], which makes the length-scales initialisation to 1 more sensible.

**Comment by vdutor** [comment]
Hi @st-- , yes this notebook takes a little bit too long to generate on the fly. I was thinking that it makes sense to have some precompiled notebooks as well in our documentation to explain some more complicated models rather than keeping it to toy examples. What do you think?

File: docs/notebooks/deep_cde.ipynb [hunk 2]
```diff
@@ -0,0 +1,643 @@
...
+    "The errorbars of the single layer model are not good: we observe an overestimation of the error bars on the left and right. "
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Deep Gaussian process with latent variables\n",
+    "\n",
+    "To tackle the problem we suggest a Deep Gaussian process with a latent variable in the first layer. The latent variable will be able to capture the \n",
```
Comment (awav) [inline docs/notebooks/deep_cde.ipynb]: A bit of word shuffling: "We suggest a Deep Gaussian process with a latent variable in the first layer to improve the error bars on the given dataset. The latent variable allows to model the heteroscedasticity, while an extra layer makes the model more expressive to catch sharp transitions."

File: docs/notebooks/deep_cde.ipynb [hunk 3]
```diff
@@ -0,0 +1,643 @@
...
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Latent Variable Layer\n",
+    "\n",
+    "This layer concatenates the inputs with a latent variable. See Dutordoir, Salimbeni et al. Conditional Density with Gaussian processes (2018) <cite data-cite=\"dutordoir2018cde\"/> for full details. We choose a one-dimensional input and a full parameterisation for the latent variables. This means that we do not need to train a recognition network, which is useful for fitting but can only be done in the case of small datasets, as is the case here."
```
Comment (awav) [inline docs/notebooks/deep_cde.ipynb:248]: Make a ref [...] instead of the long text?

File: docs/notebooks/deep_cde.ipynb [hunk 4]
```diff
@@ -0,0 +1,643 @@
...
+    }
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Fit\n",
+    "\n",
+    "We can now fit the model. Because of the `DirectlyParameterizedEncoder` it is important to set the batch size to the number of datapoints and turn off shuffle. This is so that we use the associated latent variable for each datapoint. If we would use an Amortized Encoder network this would not be necessary."
```
Comment (awav) [inline docs/notebooks/deep_cde.ipynb]: "It is important to set the batch size to the number of datapoints and turn off shuffle, checkout `DirectlyParameterizedEncoder` for details"? This needs a bit of explanation, what `DirectlyParameterizedEncoder` is.

**Review by awav — APPROVED** [review]
The result is still different comparing to Neil's post. But I guess this is okay, as the perfect match is not possible because of the difference in the modelling.

Overall, LGTM. (check the comments before merging)
